lib/between_meals/repo.rb

```diff
@@ -28,21 +32,28 @@ def initialize(repo_path, logger)
       @repo = nil
       @bin = nil
       setup
-    rescue
-      @logger.warn("Unable to read repo from #{File.expand_path(repo_path)}")
-      exit(1)
     end
 
     def self.get(type, repo_path, logger)
+      repo_path = ::Pathname.new(repo_path).realpath
       case type
+      when 'auto'
+        [
+          BetweenMeals::Repo::Git,
+          BetweenMeals::Repo::Hg,
+          BetweenMeals::Repo::Svn,
+        ].each do |repo|
+          begin
+            return repo.new(repo_path, logger)
+          rescue
```

Comment:
instead of breaking the API, check your return code here. Loop over these until you have somethign that passes `#exists?`